### PR TITLE
dlgCopySheet

### PR DESCRIPTION
**Pull request by lilyclements**
@africanmathsinitiative/instat this is ready for review

### CODE REVIEW

**Comment by lilyclements** [comment]
This code has been re-updated since the updated dialog setup and added conditions in #2618 so this is ready for review.

File: instat/dlgCopySheet.vb [hunk 1]
```diff
@@ -16,18 +16,20 @@
 Imports instat.Translations
 Public Class dlgCopySheet
     Private clsDefaultFunction As New RFunction
-    Public bFirstLoad As Boolean = True
+    Private bFirstLoad As Boolean = True
```
Comment (maxwellfundi) [inline instat/dlgCopySheet.vb]: This needs to move to SetDefaults

Comment (lilyclements) [inline instat/dlgCopySheet.vb]: Thanks! Sorted it now

Comment (maxwellfundi) [inline instat/dlgCopySheet.vb]: Great, very well done. This is now ready for you @dannyparsons

File: instat/dlgCopySheet.vb [hunk 2]
```diff
@@ -15,60 +15,82 @@
...
+    Private Sub ucrInputNewName_ContentsChanged() Handles ucrDataFrameCopySheets.ControlContentsChanged, ucrNewDataFrameName.ControlContentsChanged
         TestOKEnabled()
     End Sub
 
+    Private Sub ucrDataFrameToRename_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrDataFrameCopySheets.ControlValueChanged
+        CheckAutoName()
+    End Sub
+
+    Private Sub CheckAutoName()
+        If Not ucrNewDataFrameName.bUserTyped Then
```
Comment (dannyparsons) [inline instat/dlgCopySheet.vb]: Should also check the data frame is not empty before doing this.

**Pull request merged**